rpc/src/v1/impls/devel.rs

```diff
@@ -93,4 +105,239 @@ where
+                    metadata: format!("{:?}", Instant::now()),
+                    output: AssetMintOutput {
+                        lock_script_hash: H160::from("b042ad154a3359d276835c903587ebafefea22af"),
+                        parameters: vec![],
+                        amount: Some($amount),
+                    },
+                    approver: None,
+                    administrator: None,
+                }
+            };
+        }
+
+        macro_rules! asset_type {
+            ($tx:expr) => {
+                AssetSchemeAddress::new($tx.hash(), 0).into()
+            };
+        }
+
+        macro_rules! payment_parcel {
+            ($seq:expr, $fee:expr, $address:expr) => {
+                payment_parcel!($seq, $fee, $address, 1)
+            };
+            ($seq:expr, $fee:expr, $address:expr, $amount: expr) => {
+                Parcel {
+                    seq: $seq,
+                    fee: $fee,
+                    network_id,
+                    action: Action::Payment {
+                        receiver: $address,
+                        amount: $amount,
+                    },
+                }
+            };
+        }
+
+        macro_rules! tx_parcel {
+            ($seq:expr, $fee:expr, $transaction:expr) => {
+                Parcel {
+                    seq: $seq,
+                    fee: $fee,
+                    network_id,
+                    action: Action::AssetTransaction {
+                        transaction: $transaction,
+                        approvals: vec![],
+                    },
+                }
+            };
+        }
+
+        macro_rules! send_parcel {
+            ($parcel:expr) => {
+                let signed = SignedParcel::new_with_sign($parcel, genesis_keypair.private());
+                last_signed_parcel_hash = signed.hash();
+                self.miner.import_own_parcel(&*self.client, signed).map_err(errors::parcel_core)?;
+            };
+        }
+
+        macro_rules! transfer_input {
+            ($hash:expr, $index:expr, $asset_type:expr, $amount: expr) => {
+                AssetTransferInput {
+                    prev_out: AssetOutPoint {
+                        transaction_hash: $hash,
+                        index: $index,
+                        asset_type: $asset_type,
+                        amount: $amount,
+                    },
+                    timelock: None,
+                    lock_script: vec![0x30, 0x01],
+                    unlock_script: vec![],
+                }
+            };
+        }
+
+        macro_rules! transfer_output {
+            ($asset_type:expr, $amount:expr) => {
+                AssetTransferOutput {
+                    lock_script_hash: H160::from("b042ad154a3359d276835c903587ebafefea22af"),
+                    parameters: vec![],
+                    asset_type: $asset_type,
+                    amount: $amount,
+                }
+            };
+        }
+
+        macro_rules! tps {
```

Comment:
Why do you make it a macro instead of a function?

Reply:
Because it's hard to pass default arguments (e.g. fee, network_id) via functions. If I make them to  functions, they'll have a separated environment with `test_tps`. I think most of them are good to make as macros.

Reply:
I've moved some macros to functions.